dsp/modules/azure_openai.py

```diff
@@ -4,7 +4,16 @@
 from typing import Any, Callable, Literal, Optional, cast
 
 import backoff
-import openai
+
+try:
+    """
+    If there is any error in the langfuse configuration, it will turn to request the real address(openai or azure endpoint)
+    """
+    import langfuse
+    from langfuse.openai import openai
```

Comment:
I believe this causes dspy to use the langfuse openai wrapper if langfuse is simply installed, even if env variables are not configured. 

Considering langfuse also have a [decorator integration method](https://langfuse.com/docs/sdk/python/decorators) I think the wrapper should only be used if langfuse is configured via env variables or offer some other flag to switch of the use of the langfuse openai wrapper